server/app/views/admin/apikeys/ApiKeyNewOneView.java

```diff
@@ -0,0 +1,142 @@
+  }
+
+  public Content render(
+      Request request, ImmutableSet<String> programNames, Optional<DynamicForm> dynamicForm) {
+    String title = "Create a new API key";
+
+    ContainerTag formTag =
+        form()
+            .withMethod("POST")
+            .with(
+                makeCsrfTokenInputTag(request),
+                setStateIfPresent(
+                        FieldWithLabel.input()
+                            .setFieldName(ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                            .setId(ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                            .setLabelText("API key name"),
+                        dynamicForm,
+                        ApiKeyService.FORM_FIELD_NAME_KEY_NAME)
+                    .getContainer(),
+                h2("Expiration date"),
+                p(
+                    "Specify a date when this API key will no longer be valid. The expiration date"
+                        + " may be edited after the API is created. It is strongly recommended to"
+                        + " allow API keys to expire and create new ones to ensure that if a key"
+                        + " has been stolen, malicious users will eventually lose access. For"
+                        + " highly sensitive data, expiring keys once a week or once a month is"
+                        + " recommended. Once a year is acceptable in most situations."),
+                setStateIfPresent(
+                        FieldWithLabel.date()
+                            .setFieldName(ApiKeyService.FORM_FIELD_NAME_EXPIRATION)
+                            .setId(ApiKeyService.FORM_FIELD_NAME_EXPIRATION)
+                            .setLabelText("Expiration date"),
+                        dynamicForm,
+                        ApiKeyService.FORM_FIELD_NAME_EXPIRATION)
+                    .getContainer(),
+                h2("Allowed IP addresses"),
+                p(
+                    "Specify a subnet using CIDR notation that is allowed to call the API. The"
+                        + " subnet may be edited after the API key is created. A single IP address"
+                        + " can be specified with a mask of /32. For example, \"8.8.8.8/32\""
+                        + " allows only the IP 8.8.8.8 to use the API key. All IP addresses can be"
+                        + " allowed, but this is strongly discouraged for security reasons because"
+                        + " it would allow any computer connected to the internet to use the key"
+                        + " if they obtain it. All IP addresses can be allowed with a value of"
+                        + " 0.0.0.0/0."),
+                setStateIfPresent(
+                        FieldWithLabel.input()
+                            .setFieldName(ApiKeyService.FORM_FIELD_NAME_SUBNET)
+                            .setId(ApiKeyService.FORM_FIELD_NAME_SUBNET)
+                            .setLabelText("API key name"),
```

Comment:
I think "key name" needs updating for this field?  I get lost in the page construction so I might be off.

Reply:
done